dsgrid/registry/project_registry_manager.py

```diff
@@ -267,6 +280,114 @@ def _register_project_and_dimensions(
         self._register(config, submitter, log_message, force)
         context.add_id(RegistryType.PROJECT, config.model.project_id, self)
 
+    def _register_dimensions_from_models(
+        self,
+        src_dir,
+        dimensions: List,
+        dimension_references,
+        context,
+        submitter,
+        log_message,
+        force,
+    ):
+        dim_model = DimensionsConfigModel(dimensions=dimensions)
+        dims_config = DimensionsConfig.load_from_model(dim_model, src_dir)
+        dimension_ids = self._dimension_mgr.register_from_config(
+            dims_config, submitter, log_message, force=force, context=context
+        )
+        # Order of the next two is required for Pydantic validation.
+        dimension_references += self._dimension_mgr.make_dimension_references(dimension_ids)
+        dimensions.clear()
+
+    def _register_all_in_one_dimensions(
+        self,
+        src_dir,
+        model,
+        context,
+        submitter,
+        log_message,
+        force,
+    ):
+        new_dimensions = []
+        new_mappings = {}
+        dim_type_to_ref = {x.dimension_type: x for x in model.dimensions.base_dimension_references}
+        for dimension_type in model.dimensions.all_in_one_supplemental_dimensions:
+            if dimension_type == DimensionType.TIME:
+                raise DSGInvalidParameter("Cannot have time in all_in_one_supplemental_dimensions")
+            if dimension_type in new_mappings:
+                raise DSGInvalidParameter(
+                    f"{dimension_type} is listed twice in all_in_one_supplemental_dimensions"
+                )
+
+            dim_ref = dim_type_to_ref[dimension_type]
+            dim_config = self._dimension_mgr.get_by_id(dim_ref.dimension_id)
+            supp_dim_dir = src_dir / "dimensions" / "supplemental"
+            if not supp_dim_dir.exists():
+                raise DSGInvalidParameter(
+                    f"supplemental dimensions directory {supp_dim_dir} does not exist"
+                )
+            supp_map_dir = src_dir / "dimension_mappings" / "base_to_supplemental"
+            if not supp_map_dir.exists():
+                raise DSGInvalidParameter(
+                    f"supplemental dimension mappings directory {supp_map_dir} does not exist"
+                )
+            dt_str = dimension_type.value
+            dt_plural = f"all_{dt_str}s"
```

Comment:
There are a few items here that may need to change:
1. Do we want to generate `all_sectors` or something like `all_sector_records`? The former works as long as no dimension type ends with an `s`.
2. This introduces requirements for the project directories `dimensions/supplemental` and `dimension_mappings/base_to_supplemental`. That isn't great. An alternative is to add `ProjectConfigModel` fields that specify the names of these directories.

Reply:
Regarding 2., there can be a better solution. When it comes to the registry, these extra directories don't matter. We probably don't want to create these files in the user location because it will cause "git unknown files" if they have a repo. Maybe we can write them to a temp dir. I'll think about it.